src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java

```diff
@@ -670,13 +669,18 @@ public boolean containsInstanceField(String field) {
          *
          * @param field the field to check
          * @return true if this FieldFrame contains static field
-         * @noinspection TailRecursion
-         * @noinspectionreason TailRecursion - until issue #14814
          */
         public boolean containsStaticField(String field) {
-            return staticFields.contains(field)
-                    || parent != null
-                    && parent.containsStaticField(field);
+            FieldFrame currentParent = parent;
+            boolean parentContains = true;
+            while (currentParent != null) {
+                parentContains = parentContains && currentParent.staticFields.contains(field);
+                if (parentContains) {
+                    break;
```

Comment:
this condition can be moved into while loop, but doing that has 0.95 branch coverage ratio.

Reply:
We need to add tests to recover coverage.

Reply:
no need to worry about that now. simplied the logic